src/gt4py/next/iterator/ir_utils/common_pattern_matcher.py

```diff
@@ -32,6 +31,19 @@ def is_let(node: itir.Node) -> TypeGuard[itir.FunCall]:
     return isinstance(node, itir.FunCall) and isinstance(node.fun, itir.Lambda)
 
 
-def is_if_call(node: itir.Expr) -> TypeGuard[itir.FunCall]:
-    """Match expression of the form `if_(cond, true_branch, false_branch)`."""
-    return isinstance(node, itir.FunCall) and node.fun == im.ref("if_")
+def is_call_to(node: itir.Node, fun: str | list[str]) -> TypeGuard[itir.FunCall]:
+    """
+    Match call expression to a given function.
+
+    >>> from gt4py.next.iterator.ir_utils import ir_makers as im
+    >>> node = im.call("make_tuple")(1, 2)
+    >>> is_call_to(node, "make_tuple")
+    True
+    >>> is_call_to(node, "plus")
+    False
+    """
+    if isinstance(fun, (list, tuple, set)):
```

Comment:
This `isinstance()` check doesn't match the typying annotation for `fun`.

Reply:
I switched to `collections.abc.Iterable`, but still included `list`, `tuple`, `set` for performance reasons.